File: refactoring-practice-1/src/main/typescript/replace_parameter_with_method/price-calculator.ts
export class PriceCalculator {

    public constructor(private unitPrice: number, private quantity: number) {
    }

    public getPrice(): number {
        let basePrice = this.quantity * this.unitPrice;
        let discountLevel;
        if (this.quantity > 100) {
            discountLevel = 2;
        } else {
            discountLevel = 1;
        }
        let finalPrice = this.getDiscountedPrice(basePrice, discountLevel);
        return finalPrice;
    }

    /*
     * Ebből a metódusból távolítsuk el mindkét paramétert úgy, hogy helyettesítjük
     * őket egy getBasePrice és egy getDiscountLevel metódushívással!
     */
    private getDiscountedPrice(basePrice: number, discountLevel: number): number {
        if (discountLevel == 1) {
            return basePrice * 0.1;
        } else {
            return basePrice * 0.05;
        }
    }

}

File: refactoring-practice-1/src/main/java/dojo/replace_parameter_with_method/PriceCalculator.java
package dojo.replace_parameter_with_method;

public class PriceCalculator {

	private final double unitPrice;
	private final int quantity;
	
	public PriceCalculator(double unitPrice, int quantity) {
		super();
		this.unitPrice = unitPrice;
		this.quantity = quantity;
	}
	
	public double getPrice() {
		double basePrice = quantity * unitPrice;
		int discountLevel;
		if (quantity > 100) {
			discountLevel = 2;
		} else {
			discountLevel = 1;
		}
		double finalPrice = getDiscountedPrice(basePrice, discountLevel);
		return finalPrice;
	}
	
	/*
	 * Ebből a metódusból távolítsuk el mindkét paramétert úgy, hogy helyettesítjük
	 * őket egy getDiscountLevel és egy getBasePrice metódushívással!
	 * A refaktorálás végén ezt a metódust akár be is olvaszthatjuk a getPrice metódusba.
	 */
	private double getDiscountedPrice(double basePrice, int discountLevel) {
		if (discountLevel == 1) {
			return basePrice * 0.1;
		} else {
			return basePrice * 0.05;
		}
	}
	
}
